code/game/turfs/open/floor/catwalk_plating.dm

```diff
@@ -16,24 +16,28 @@
 	overfloor_placed = TRUE
 	underfloor_accessibility = UNDERFLOOR_VISIBLE
 	var/covered = TRUE
+	var/above_state = "maint_above"	//Icon-state for the overlay
 
 
 /turf/open/floor/catwalk_floor/Initialize(mapload)
 	. = ..()
 	update_icon(UPDATE_OVERLAYS)
 
+GLOBAL_LIST_INIT(catwalk_overlay_masterlist, list())	//Stores all the above_states for the different types of catwalk
```

Comment:
Stores all the different types of overlays for the catwalk, right? Not just all the icon states.

You may want `GLOBAL_LIST_EMPTY` here too instead of `GLOBAL_LIST_INIT`, it gets rid of the need for that second argument

Reply:
yeah, it stores the icon_state paired with the overlay variables, so all you need to do is select it and it'll be on the proper layer and everything
at least, thats the intention. it seems to work that way too, so i HOPEFULLY did it right